messaging/src/main/java/org/axonframework/commandhandling/distributed/DistributedCommandBus.java

```diff
@@ -177,25 +158,26 @@ public <C, R> void dispatch(@Nonnull CommandMessage<C> command,
         CommandMessage<? extends C> interceptedCommand = intercept(command);
         MessageMonitor.MonitorCallback messageMonitorCallback = messageMonitor.onMessageIngested(interceptedCommand);
         Optional<Member> optionalDestination = commandRouter.findDestination(interceptedCommand);
-        if (optionalDestination.isPresent()) {
-            Member destination = optionalDestination.get();
-            try {
+        Span span = spanFactory.createDispatchSpan(() -> "DistributedCommandBus.dispatch", command).start();
+        try (SpanScope ignored = span.makeCurrent()) {
+            if (optionalDestination.isPresent()) {
+                Member destination = optionalDestination.get();
                 connector.send(destination,
                                interceptedCommand,
                                new MonitorAwareCallback<>(callback, messageMonitorCallback));
-            } catch (Exception e) {
-                messageMonitorCallback.reportFailure(e);
-                destination.suspect();
-                callback.onResult(interceptedCommand, asCommandResultMessage(
-                        new CommandDispatchException(DISPATCH_ERROR_MESSAGE + ": " + e.getMessage(), e)
-                ));
+                span.end();
```

Comment:
Ending the span while the span is the current should not be done. This should be moved outside of the try-block, or the SpanFactory will log an error to be reported to us (could be a potential leak - span is ended, but it's still the active for the thread)

Reply:
Moved to finally block.